mllib/src/main/scala/org/apache/spark/ml/tuning/ValidatorParams.scala

```diff
@@ -183,8 +195,14 @@ private[ml] object ValidatorParams {
           val paramPairs = pMap.map { case pInfo: Map[String, String] =>
             val est = uidToParams(pInfo("parent"))
             val param = est.getParam(pInfo("name"))
-            val value = param.jsonDecode(pInfo("value"))
-            param -> value
+            if (pInfo("isJson").toBoolean.booleanValue()) {
+              val value = param.jsonDecode(pInfo("value"))
+              param -> value
+            } else {
+              val path = param.jsonDecode(pInfo("value")).toString
+              val value = DefaultParamsReader.loadParamsInstance[MLWritable](path, sc)
```

Comment:
This is OK with me for now since it will address all cases I've seen.  In the future, it'd be great to make this more general by allowing it to read any MLReadable type (not just DefaultParamsReadable).  I'll comment in the save() section above about this too.